src/components/academy/grading/GradingEditor.tsx

```diff
@@ -1,36 +1,42 @@
-import { ButtonGroup, Icon, NumericInput, Position } from '@blueprintjs/core'
+import { ButtonGroup, Icon, Intent, NumericInput, Position } from '@blueprintjs/core'
 import { IconNames } from '@blueprintjs/icons'
 import * as React from 'react'
 import ReactMde, { ReactMdeTypes } from 'react-mde'
+import { Prompt } from 'react-router'
 import * as Showdown from 'showdown'
+
 import { controlButton } from '../../commons'
 
-type GradingEditorProps = DispatchProps & OwnProps & StateProps
+type GradingEditorProps = DispatchProps & OwnProps
 
 export type DispatchProps = {
-  handleGradingCommentsChange: (s: string) => void
-  handleGradingXPChange: (i: number | undefined) => void
-  handleGradingInputSave: (s: string, i: number | undefined) => void
+  handleGradingSave: (
+    submissionId: number,
+    questionId: number,
+    g: number,
+    s: string,
+    i: number | undefined
+  ) => void
```

Comment:
Full param names (since prettier put them on separate lines already). I think abbreviated param names is acceptable just for stuff like the tests.